server/src/main.rs

```diff
@@ -279,9 +289,9 @@ fn main() {
                     _ = signal::sighup_compact() => {},
                     _ = signal::sigusr1_dump_config() => {},
                     _ = signal::sigusr2_tokio_dump() => {},
-                    _ = task_center_watch.cancelled() => {
-                        shutdown = true;
+                    _ = &mut tc_cancelled => {
                         // Shutdown was requested by task center and it has completed.
+                        break;
```

Comment:
If a `TaskKind` with `OnError = "shutdown"` fails, then the main thread will break here, right? If yes, would we then miss running `on_ungraceful_shutdown` because we complete the future with `()`?

Reply:
Addressed in the latest changes.